lib/cucumber/runtime.rb

```diff
@@ -228,6 +228,12 @@ def legacy_formatter?(factory)
     def failure?
       if @configuration.wip?
         summary_report.test_cases.total_passed > 0
+      elsif @configuration.retry_attempts > 0
+        unless summary_report.test_cases.total_passed != @configuration.total_cases
```

Comment:
Golfing the code: 

- an `unless` with `!=` could become an `if` with a `==`

Would that read better?